test/common/quic/envoy_quic_client_session_test.cc

```diff
@@ -551,5 +555,38 @@ TEST_P(EnvoyQuicClientSessionTest, UsesUdpGro) {
                       dispatcher_->run(Event::Dispatcher::RunType::RunUntilExit));
 }
 
+// Ensures that the Network::Utility::readFromSocket function does uses recvmmsg for client
+// QUIC connections when GRO is not supported.
+TEST_P(EnvoyQuicClientSessionTest, UsesRecvMmsgWithoutGro) {
+  NiceMock<MockOsSysCallsImpl> os_sys_calls_;
```

Comment:
Please check the flag value and skip test if it is false.

Ditto to the test above.